src/ip_address.h

```diff
@@ -101,6 +111,11 @@ class IPAddress {
         return m_addr6;
     }
 
+    inline const char *addr_un() const
+    {
+        return m_addr_un;
```

Comment:
It's dangerous because `m_addr_un` may not contain trailing zero.

Reply:
Not sure how to approach this issue

Reply:
The simplest way is to use `std::string` as a storage.

Reply:
string * with new/delete
or char* with strdup and free?

Reply:
I propose to use just a member of type std::string outside the `union`. This will slightly increase structure size but not so dramatically as for char[MAX_UDS_NAME].